wordpress-mobile/WordPress-iOS
# [Stats Refresh] Add Period > Referrers card

Fixes #10998 

This add the Period Referrers card. Deets:
- Shows top 6 referrers, with expanding rows where necessary.
- If a row cannot be expanded, selecting it will launch a web view.
- Selecting an expandable row will show up to 10 child rows.
- If a _child_ row can be expanded, the label color will remain dark grey.
- If a _child_ row cannot be expanded, the label color will be light grey.
- Expanding a _child_ row: (NOTE: this seems to only be applicable to `Search Engines`)
  - Max 10 of its child rows will be displayed.
  - The label color will be light grey.
  - Full width separator lines appear above and below the section.

Notes:
- There is not a default placeholder icon, i.e. when a Referrer icon has not loaded the image view will be blank. I'll add the placeholder once @SylvesterWilmott provides it. (#11080)
- The method I implemented to keep track of expanded rows has a flaw, and can have duplicate entries. 🤦‍♂️ So, if you expand, say this one, both will be expanded. I'll fix that in a future PR (https://github.com/wordpress-mobile/WordPress-iOS/pull/11081).
<img width="300" alt="exp_rows" src="https://user-images.githubusercontent.com/1816888/52973689-199a1380-337c-11e9-9e59-196c93d04dc6.png">


To test:

---
- On a site with Referrers, go to Stats > Period > Referrers. (en.blog is a good one)
- Verify the initial view is:
<img width="350" alt="referrers_init" src="https://user-images.githubusercontent.com/1816888/52971933-74c90780-3376-11e9-8f5a-3c91c49b91c7.png">

---
- Select an expandable row.
- Verify the view is:
<img width="350" alt="referrers_exp_once" src="https://user-images.githubusercontent.com/1816888/52971991-a346e280-3376-11e9-97d1-990a38b29523.png">

---
- Select an expandable row within an expanded section.
- Verify the view is:
<img width="350" alt="referrers_exp_twice" src="https://user-images.githubusercontent.com/1816888/52972100-f4ef6d00-3376-11e9-8bd4-1b1f53df40a2.png">

---
- On a site with no Referrers, verify the view is:
<img width="350" alt="referrers_no_data" src="https://user-images.githubusercontent.com/1816888/52972145-19e3e000-3377-11e9-928e-22513737e4ab.png">
